app/mailers/twilio_sender.rb

```diff
@@ -46,18 +98,38 @@ def self.get_phone_numbers_from_flow_execution(execution_id)
       Rails.logger.warn e.error_message
       return
     end
-
     # Get a message out of the studio execution which we can get the To/From numbers out of
     # The opt-in/out could come from an incoming message trigger OR an existing execution
     # The message pulled from an existing execution will be the first inbound message found within the execution
     message = execution&.context&.[]('trigger')&.[]('message') || execution&.context&.[]('widgets')&.values&.select do |x|
                                                                     x&.[]('inbound')
                                                                   end&.[](0)&.[]('inbound')
-    phone_number_from = message&.[]('From')
-    phone_number_to = message&.[]('To')
+    # Alternatively, the execution could be of type SINGLE_SMS where we sent a single outbound message
+    message = execution&.context&.[]('widgets')&.values&.select { |x| x&.[]('To') }&.[](0) if message.nil?
+
+    return nil if message.nil?
+
+    if !message&.[]('outbound').nil? || (message&.[]('Direction')&.include? 'outbound')
+      phone_number_from = message&.[]('To')
+      phone_number_to = message&.[]('From')
+    else
+      phone_number_from = message&.[]('From')
+      phone_number_to = message&.[]('To')
+    end
 
     return { monitoree_number: phone_number_from, sara_number: phone_number_to } if !phone_number_from.nil? && !phone_number_to.nil?
 
     nil
   end
+
+  def self.dispatch_errored_contact_history_items(patient, error_message)
+    pats = if patient&.responder == patient
+             # If errored contact was for a communication for all dependents ie: sms_assessment or voice_assessment
```

Comment:
How does the check on responder tell you if the error was on sms_assessment or voice_assessment vs weblink? Couldn't there be an error on the weblink sent to the HoH and this would be true?

Reply:
Good question. The only case where an error would come back from Twilio with a patient_submission_token that belongs to a non-HoH is in the case of a weblink since the patient_submission_token for voice and SMS assessments is always belongs to the HoH.  However, as I am reading this now I realize that in the event of a failed HoH weblink, the dependents would also get error message. I will refactor this to check for preferred contact method